services/src/main/scala/cromwell/pipeline/service/UserService.scala

```diff
@@ -86,14 +120,24 @@ object UserService {
       }
 
       private def getUserById(userId: UserId): Future[UserWithCredentials] =
-        userRepository.getUserById(userId).flatMap {
-          case Some(user) => Future.successful(user)
-          case None       => Future.failed(new RuntimeException("user with this id doesn't exist"))
-        }
+        userRepository
+          .getUserById(userId)
+          .flatMap {
+            case Some(user) => Future.successful(user)
+            case None       => Future.failed(NotFound())
+          }
+          .recoverWith {
+            case exc: ServiceException => Future.failed(exc)
+            case _: Exception =>
+              Future.failed(InternalError("Failed to find user due to unexpected internal error"))
+          }
```

Comment:
How about `transform` ?

Reply:
I didn't manage to use this method. Coud you explain please, how I can use it.

Reply:
Oh. I got it. Now it's changed :)